src/components/input/index.tsx

```diff
@@ -120,7 +120,7 @@ const Input = forwardRef<HTMLInputElement, Props>(
       return () => {
         debouncedChangeHandler?.cancel()
       }
-    }, [])
+    }, [debouncedChangeHandler])
```

Comment:
debouncedChangeHandler changes everytime the `onChange` function changes. As a result, you should check if all components have this function in a callback

Reply:
have you checked if the number of times the hook runs is fine?

Reply:
looks okeyish